sherpa/astro/data.py

```diff
@@ -5379,6 +5425,66 @@ def filter_region(self, data):
 
 
 class DataIMGInt(DataIMG):
+    '''Binned image data set, including functions for coordinate transformations
+
+    Parameters
+    ----------
+    name : string
+        name of this dataset
+    x0lo : array-like
+        Lower bounds of the bins for coordinates in one dimension
+    x1lo : array-like
+        Lower bound of the bins for coordinates in the other dimension
+    x0hi : array-like
+        Upper bound of the bins for coordinates in one dimension
+    x1hi : array-like
+        Upper bound of the bins for coordinates in the other dimension
+    y : array-like
+        The values of the dependent observable.
+    shape : tuple
+        Shape of the data grid for regularly gridded data (optional). This is
+        used return the data as an image e.g. for display,
+        but it not needed for fitting and modelling.
+    staterror : array-like
+        the statistical error associated with the data
+    syserror : array-like
+        the systematic error associated with the data
+    sky : `sherpa.astro.io.wcs.WCS`
+        The optional WCS object that converts to the physical coordinate system.
+    eqpos : `sherpa.astro.io.wcs.WCS`
+        The optional WCS object that converts to the world coordinate system.
+    coord : string
+        The coordinate system of the data. This can be one of 'logical',
+        'physical', or 'world'.
+        'image' is an alias for 'logical' and 'wcs' is an alias for 'world'.
+    header : dict
+        The FITS header associated with the data to store meta data.
+
+    Example
+    -------
+    In this example, we first generate a 1000 (x,y) points from a 2D Gaussian.
+    This could be, e.g., photons observed from a star. In x direction, the center
+    of the Gaussian is at 1.2 and in y direction at 0.0. We then use
+    `~numpy.histogram2d` to bin the data into a 2D histogram. Note that numpy's
+    convention for the ordering of the axis is not the same as Sherpa's.
+    We want to use the bin edges when we create the `DataIMGInt` object, but numpy
+    gives us just one set of edges. We therefore first pick either the upper or lower
+    edges (``x0edges[:-1]`` or ``x0edges[1:]``) and then repeat them to match
+    the number of bins in the flattened data array::
+
+        >>> import numpy as np
+        >>> x = np.random.normal(size=1000, loc=1.2)
+        >>> y = np.random.normal(size=1000)
+        >>> xrange = np.arange(-2, 4.1, 0.5)
+        >>> yrange = np.arange(-2, 2.1, 0.5)
+        >>> hist, x0edges, x1edges = np.histogram2d(y, x, bins=(yrange, xrange))
+        >>> x0lo, x1lo = np.meshgrid(x0edges[:-1], x1edges[:-1])
+        >>> x0hi, x1hi = np.meshgrid(x0edges[1:], x1edges[1:])
```

Comment:
In this example, `x0lo` ... etc that I pass in are effectively the physical coordinate system, not the logical (they are not integers from 0..n_pixel). That's not an uncommon use case and, in the absence of WCS in this example make sense to me. But is it confusing as an example?

I wrote is like this because I think it explains the use case and then, once I understand #1786 , I can add another example *with* a WCS and explain the difference.
However, I feel it's better to get a short PR with these docs in now then to wait for something in the future.